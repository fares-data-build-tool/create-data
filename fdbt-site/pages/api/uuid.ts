import { NextApiRequest, NextApiResponse } from 'next'
import { v1 } from 'uuid'

export default (req: NextApiRequest, res: NextApiResponse) => {
    console.log("uuid endpoint has been called");
    const uuid = v1();
    const objectToReturn = { uuid };
    res.status(200).json(objectToReturn);
}