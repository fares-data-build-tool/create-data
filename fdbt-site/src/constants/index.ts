export const OPERATOR_COOKIE = 'fdbt-operator';

export const FARE_TYPE_COOKIE = 'fdbt-fare-type';

export const PASSENGER_TYPE_COOKIE = 'fdbt-passenger-type';

export const SERVICE_COOKIE = 'fdbt-service';

export const JOURNEY_COOKIE = 'fdbt-journey';

export const GOVUK_LINK = 'https://www.nationalarchives.gov.uk/doc/open-government-licence/version/3/';

export const FARE_STAGES_COOKIE = 'fdbt-fare-stages';

export const STAGE_NAME_VALIDATION_COOKIE = 'fdbt-stage-names-validation';

export const FEEDBACK_LINK = 'mailto:fdbt@transportforthenorth.com?bcc=tfn-feedback@infinityworks.com';

export const STAGE_NAMES_COOKIE = 'fdbt-stage-names';

export const DAYS_VALID_COOKIE = 'fdbt-days-valid';

export const PRODUCT_DETAILS_COOKIE = 'fdbt-product-details';

export const PERIOD_TYPE_COOKIE = 'fdbt-period-type';

export const CSV_UPLOAD_COOKIE = 'fdbt-csv-upload';

export const CSV_ZONE_UPLOAD_COOKIE = 'fdbt-csv-zone-upload';

export const PERIOD_EXPIRY_COOKIE = 'fdbt-period-expiry';

export const SERVICE_LIST_COOKIE = 'fdbt-services';

export const INPUT_METHOD_COOKIE = 'fdbt-input-method';

export const NUMBER_OF_STAGES_COOKIE = 'fdbt-number-stages';

export const MATCHING_COOKIE = 'fdbt-matching';

export const MULTIPLE_SERVICE_COOKIE = 'fdbt-multiple-service';

export const NUMBER_OF_PRODUCTS_COOKIE = 'fdbt-number-of-products';

export const MULTIPLE_PRODUCT_COOKIE = 'fdbt-multiple-product';

export const ALL_COOKIES: string[] = [
    'fdbt-operator',
    'fdbt-fare-type',
    'fdbt-passenger-type',
    'fdbt-service',
    'fdbt-journey',
    'fdbt-fare-stages',
    'fdbt-stage-names-validation',
    'fdbt-stage-names',
    'fdbt-price-entry',
    'fdbt-days-valid',
    'fdbt-product-details',
    'fdbt-period-type',
    'fdbt-csv-upload',
    'fdbt-csv-zone-upload',
    'fdbt-number-of-products',
    'fdbt-period-expiry',
    'fdbt-services',
    'fdbt-input-method',
    'fdbt-matching',
    'fdbt-number-stages',
    'fdbt-multiple-service',
    'fdbt-number-of-products',
    'fdbt-multiple-product',
];

export const ALLOWED_CSV_FILE_TYPES = [
    'text/plain',
    'text/x-csv',
    'application/vnd.ms-excel',
    'application/csv',
    'application/x-csv',
    'application/octet-stream',
    'text/csv',
    'text/comma-separated-values',
    'text/x-comma-separated-values',
    'text/tab-separated-values',
];

export const STAGE = process.env.STAGE || 'dev';

export const RAW_USER_DATA_BUCKET_NAME = `fdbt-raw-user-data-${STAGE}`;
export const USER_DATA_BUCKET_NAME = `fdbt-user-data-${STAGE}`;
export const MATCHING_DATA_BUCKET_NAME = `fdbt-matching-data-${STAGE}`;
