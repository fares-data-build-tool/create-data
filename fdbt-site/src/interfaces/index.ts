export interface PeriodProductType {
    productName: string;
    productPrice: string;
    productNameError: boolean;
    productPriceError: boolean;
}
