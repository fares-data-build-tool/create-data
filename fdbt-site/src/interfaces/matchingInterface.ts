import { Stop } from '../data/auroradb';

export interface Price {
    price: string;
    fareZones: string[];
}

export interface MatchingFareZonesData {
    name: string;
    stops: Stop[];
    prices: Price[];
}

export interface MatchingFareZones {
    [key: string]: MatchingFareZonesData;
}
