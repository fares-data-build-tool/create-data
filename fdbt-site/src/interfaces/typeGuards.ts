import {
    SingleTicketProps,
    ReturnTicketProps,
    PeriodTicketProps,
    FlatFareTicketProps,
} from '../pages/ticketConfirmation';
import { SelectSalesOfferPackageWithError } from '../pages/api/selectSalesOfferPackage';
import {
    PeriodTicket,
    PointToPointTicket,
    PeriodMultipleServicesTicket,
    FlatFareTicket,
    GeoZoneTicket,
    InputMethodInfo,
    ErrorInfo,
    Journey,
    JourneyWithErrors,
    TicketRepresentationAttribute,
    TicketRepresentationAttributeWithErrors,
    SalesOfferPackage,
    ProductWithSalesOfferPackages,
    TicketPeriod,
    MultiOperatorInfo,
    MultiOperatorInfoWithErrors,
} from './index';

import { FareType, FareTypeWithErrors } from '../pages/api/fareType';
import { PassengerType, PassengerTypeWithErrors } from '../pages/api/passengerType';
import { Service, ServiceWithErrors } from '../pages/api/service';
import { FareStagesAttribute, FareStagesAttributeWithErrors } from '../pages/api/chooseStages';
import { InputCheck } from '../pages/stageNames';
import { TicketPeriodWithErrors } from '../pages/api/productDateInformation';
import { MultipleOperatorsAttribute, MultipleOperatorsAttributeWithErrors } from '../pages/api/searchOperators';

export const isNotEmpty = <T>(value: T | null | undefined): value is T => value !== null && value !== undefined;

export const isPeriodTicket = (ticket: PeriodTicket | PointToPointTicket): ticket is PeriodTicket =>
    (ticket as PeriodTicket).products?.[0]?.productName !== undefined;

export const isMultipleServicesTicket = (
    ticket: PeriodTicket | PointToPointTicket,
): ticket is PeriodMultipleServicesTicket | FlatFareTicket =>
    (ticket as PeriodMultipleServicesTicket).selectedServices !== undefined;

export const isPointToPointTicket = (ticket: PeriodTicket | PointToPointTicket): ticket is PointToPointTicket =>
    (ticket as PointToPointTicket).lineName !== undefined;

export const isGeoZoneTicket = (ticket: PeriodTicket | PointToPointTicket): ticket is GeoZoneTicket =>
    (ticket as GeoZoneTicket).zoneName !== undefined;

export const isFareTypeAttributeWithErrors = (
    fareTypeAttribute: FareType | FareTypeWithErrors,
): fareTypeAttribute is FareTypeWithErrors => (fareTypeAttribute as FareTypeWithErrors).errors !== undefined;

export const isFareType = (fareType: FareType | FareTypeWithErrors | undefined): fareType is FareType => {
    return fareType !== undefined && (fareType as FareType).fareType !== undefined;
};

export const inputMethodErrorsExist = (
    inputMethodAttribute: InputMethodInfo | ErrorInfo | undefined,
): inputMethodAttribute is ErrorInfo => (inputMethodAttribute as ErrorInfo)?.errorMessage !== undefined;

export const isPassengerTypeAttributeWithErrors = (
    fareTypeAttribute: PassengerType | PassengerTypeWithErrors | undefined,
): fareTypeAttribute is PassengerTypeWithErrors => (fareTypeAttribute as PassengerTypeWithErrors).errors !== undefined;

export const isPassengerType = (
    passengerType: PassengerType | PassengerTypeWithErrors | undefined,
): passengerType is PassengerType => {
    return passengerType !== undefined && (passengerType as PassengerType).passengerType !== undefined;
};

export const isServiceAttributeWithErrors = (
    serviceAttribute: Service | ServiceWithErrors,
): serviceAttribute is ServiceWithErrors => (serviceAttribute as ServiceWithErrors).errors !== undefined;

export const isService = (service: Service | ServiceWithErrors | undefined): service is Service => {
    return service !== undefined && (service as Service).service !== undefined;
};

export const isJourneyAttributeWithErrors = (
    journeyAttribute: Journey | JourneyWithErrors,
): journeyAttribute is JourneyWithErrors => (journeyAttribute as JourneyWithErrors).errors !== undefined;

export const isJourney = (journey: Journey | JourneyWithErrors | undefined): journey is Journey => {
    return (
        journey !== undefined &&
        ((journey as Journey).directionJourneyPattern !== undefined ||
            (journey as Journey).inboundJourney !== undefined ||
            (journey as Journey).outboundJourney !== undefined ||
            (journey as Journey).errors !== undefined)
    );
};

export const isTicketRepresentationWithErrors = (
    ticketType: TicketRepresentationAttribute | TicketRepresentationAttributeWithErrors | undefined,
): ticketType is TicketRepresentationAttributeWithErrors =>
    ticketType !== undefined && (ticketType as TicketRepresentationAttributeWithErrors).errors !== undefined;

export const isTicketRepresentation = (
    ticketType: TicketRepresentationAttribute | TicketRepresentationAttributeWithErrors | undefined,
): ticketType is TicketRepresentationAttribute =>
    ticketType !== undefined && (ticketType as TicketRepresentationAttribute).name !== undefined;

export const isInputCheck = (stageNamesInfo: string[] | InputCheck[] | undefined): stageNamesInfo is InputCheck[] =>
    stageNamesInfo !== undefined && (stageNamesInfo[0] as InputCheck).error !== undefined;

export const isFareStageWithErrors = (
    fareStages: FareStagesAttribute | FareStagesAttributeWithErrors | undefined,
): fareStages is FareStagesAttributeWithErrors =>
    fareStages !== undefined && (fareStages as FareStagesAttributeWithErrors).errors !== undefined;

export const isFareStage = (
    fareStages: FareStagesAttribute | FareStagesAttributeWithErrors | undefined,
): fareStages is FareStagesAttribute =>
    fareStages !== undefined && (fareStages as FareStagesAttribute).fareStages !== undefined;

export const isSalesOfferPackageWithErrors = (
    salesOfferPackageInfo:
        | SelectSalesOfferPackageWithError
        | SalesOfferPackage[]
        | ProductWithSalesOfferPackages[]
        | undefined,
): salesOfferPackageInfo is SelectSalesOfferPackageWithError =>
    salesOfferPackageInfo !== undefined &&
    (salesOfferPackageInfo as SelectSalesOfferPackageWithError).errors !== undefined;

export const isProductWithSalesOfferPackages = (
    salesOfferPackageInfo:
        | SelectSalesOfferPackageWithError
        | SalesOfferPackage[]
        | ProductWithSalesOfferPackages[]
        | undefined,
): salesOfferPackageInfo is ProductWithSalesOfferPackages[] =>
    salesOfferPackageInfo !== undefined &&
    (salesOfferPackageInfo as ProductWithSalesOfferPackages[])[0].productName !== undefined;

export const isSalesOfferPackages = (
    salesOfferPackageInfo:
        | SelectSalesOfferPackageWithError
        | SalesOfferPackage[]
        | ProductWithSalesOfferPackages[]
        | undefined,
): salesOfferPackageInfo is SalesOfferPackage[] =>
    salesOfferPackageInfo !== undefined &&
    (salesOfferPackageInfo as SalesOfferPackage[])[0].ticketFormats !== undefined;

export const isSingleTicketProps = (
    props: SingleTicketProps | ReturnTicketProps | PeriodTicketProps | FlatFareTicketProps,
): props is SingleTicketProps => (props as SingleTicketProps).journeyDirection !== undefined;

export const isReturnTicketProps = (
    props: SingleTicketProps | ReturnTicketProps | PeriodTicketProps | FlatFareTicketProps,
): props is ReturnTicketProps => (props as ReturnTicketProps).circular !== undefined;

export const isPeriodTicketProps = (
    props: SingleTicketProps | ReturnTicketProps | PeriodTicketProps | FlatFareTicketProps,
): props is PeriodTicketProps => (props as PeriodTicketProps).zone !== undefined;

export const isFlatFareTicketProps = (
    props: SingleTicketProps | ReturnTicketProps | PeriodTicketProps | FlatFareTicketProps,
): props is FlatFareTicketProps => (props as FlatFareTicketProps).productName !== undefined;

export const isTicketPeriodAttributeWithErrors = (
    productDates: TicketPeriod | TicketPeriodWithErrors | undefined,
): productDates is TicketPeriodWithErrors =>
    productDates !== undefined && (productDates as TicketPeriodWithErrors).errors !== undefined;

export const isTicketPeriodAttribute = (
    productDates: TicketPeriod | TicketPeriodWithErrors | undefined,
): productDates is TicketPeriod =>
    productDates !== undefined &&
    (productDates as TicketPeriod).startDate !== undefined &&
    (productDates as TicketPeriod).endDate !== undefined;

export const isMultipleOperatorAttributeWithErrors = (
    searchOperator: MultipleOperatorsAttribute | MultipleOperatorsAttributeWithErrors | undefined,
): searchOperator is MultipleOperatorsAttributeWithErrors =>
    searchOperator !== undefined && (searchOperator as MultipleOperatorsAttributeWithErrors).errors !== undefined;

export const isMultiOperatorInfoWithErrors = (
    multiOperatorInfo: MultiOperatorInfo[] | MultiOperatorInfoWithErrors | undefined,
): multiOperatorInfo is MultiOperatorInfoWithErrors =>
    multiOperatorInfo !== undefined && (multiOperatorInfo as MultiOperatorInfoWithErrors).errors !== undefined;
