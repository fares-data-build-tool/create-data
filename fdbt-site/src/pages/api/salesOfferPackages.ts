import { NextApiResponse } from 'next';
import isArray from 'lodash/isArray';
import { redirectTo, redirectToError } from './apiUtils';
import { updateSessionAttribute } from '../../utils/sessions';
import { ErrorInfo, NextApiRequestWithSession } from '../../interfaces';
import { SOP_INFO_ATTRIBUTE } from '../../constants';
import { purchaseLocationsList, paymentMethodsList, ticketFormatsList } from '../salesOfferPackages';

export interface SalesOfferPackageInfo {
    purchaseLocations: string[];
    paymentMethods: string[];
    ticketFormats: string[];
}

export interface SalesOfferPackageInfoWithErrors extends SalesOfferPackageInfo {
    errors: ErrorInfo[];
}

export default (req: NextApiRequestWithSession, res: NextApiResponse): void => {
    const errors: ErrorInfo[] = [];

    try {
        let { purchaseLocations, paymentMethods, ticketFormats } = req.body;

        if (!purchaseLocations) {
            errors.push({
                errorMessage: 'Select at least one ticket purchase location',
                id: purchaseLocationsList.id,
            });
        }

        if (!paymentMethods) {
            errors.push({
                errorMessage: 'Select at least one ticket payment method',
                id: paymentMethodsList.id,
            });
        }

        if (!ticketFormats) {
            errors.push({
                errorMessage: 'Select at least one ticket media format',
                id: ticketFormatsList.id,
            });
        }

        if (purchaseLocations && !isArray(purchaseLocations)) {
            purchaseLocations = purchaseLocations.split();
        }

        if (paymentMethods && !isArray(paymentMethods)) {
            paymentMethods = paymentMethods.split();
        }

        if (ticketFormats && !isArray(ticketFormats)) {
            ticketFormats = ticketFormats.split();
        }

        const salesOfferPackageInfo: SalesOfferPackageInfo = {
            purchaseLocations: purchaseLocations || [],
            paymentMethods: paymentMethods || [],
            ticketFormats: ticketFormats || [],
        };

        if (errors.length > 0) {
            (salesOfferPackageInfo as SalesOfferPackageInfoWithErrors).errors = errors;
            updateSessionAttribute(req, SOP_INFO_ATTRIBUTE, salesOfferPackageInfo);

            redirectTo(res, '/salesOfferPackages');
            return;
        }

        updateSessionAttribute(req, SOP_INFO_ATTRIBUTE, salesOfferPackageInfo);

        redirectTo(res, '/describeSalesOfferPackage');
    } catch (err) {
        const message = 'There was a problem in the sales offer package API.';
        redirectToError(res, message, err);
    }
};
