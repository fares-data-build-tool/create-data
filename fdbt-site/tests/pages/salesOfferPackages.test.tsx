import * as React from 'react';
import { shallow } from 'enzyme';
import SalesOfferPackages, { getServerSideProps, SalesOfferPackagesProps } from '../../src/pages/salesOfferPackages';
import { getMockContext } from '../testData/mockData';
import { ErrorInfo } from '../../src/interfaces';
import { SOP_INFO_ATTRIBUTE } from '../../src/constants';
import { SalesOfferPackageInfo, SalesOfferPackageInfoWithErrors } from '../../src/pages/api/salesOfferPackages';

describe('pages', () => {
    afterEach(() => {
        jest.resetAllMocks();
    });

    const salesOfferPackagesNoError: SalesOfferPackagesProps = {
        salesOfferPackage: {
            purchaseLocations: [],
            paymentMethods: [],
            ticketFormats: [],
        },
    };

    const salesOfferPackageWithError: SalesOfferPackagesProps = {
        salesOfferPackage: {
            purchaseLocations: [],
            paymentMethods: [],
            ticketFormats: [],
            errors: [{ errorMessage: 'error', id: '' }],
        },
    };

    describe('salesOfferPackage', () => {
        it('should render correctly', () => {
            // eslint-disable-next-line react/jsx-props-no-spreading
            const tree = shallow(<SalesOfferPackages {...salesOfferPackagesNoError} csrfToken="" pageProps={[]} />);
            expect(tree).toMatchSnapshot();
        });

        it('should render an error when errors are passed through', () => {
            // eslint-disable-next-line react/jsx-props-no-spreading
            const tree = shallow(<SalesOfferPackages {...salesOfferPackageWithError} csrfToken="" pageProps={[]} />);
            expect(tree).toMatchSnapshot();
        });
    });

    describe('getServerSideProps', () => {
        it('should show the page correctly when there is no salesPackageOffer session', () => {
            const ctx = getMockContext();
            const expectedProps: SalesOfferPackageInfo = {
                purchaseLocations: [],
                paymentMethods: [],
                ticketFormats: [],
            };

            const result = getServerSideProps(ctx).props.salesOfferPackage;
            expect(result).toEqual(expectedProps);
        });

        it('should set the select purchaseLocations if item has been selected and populate errors if two other sections not selected', () => {
            const errors: ErrorInfo[] = [
                { errorMessage: 'Select ticket Payments', id: 'paymentMethods' },
                { errorMessage: 'Select ticket formats', id: 'ticketFormats' },
            ];

            const ctx = getMockContext({
                session: {
                    [SOP_INFO_ATTRIBUTE]: {
                        purchaseLocations: ['OnBoard'],
                        paymentMethods: [],
                        ticketFormats: [],
                        errors,
                    },
                },
            });

            const expectedProps: SalesOfferPackageInfoWithErrors = {
                purchaseLocations: ['OnBoard'],
                paymentMethods: [],
                ticketFormats: [],
                errors,
            };

            const result = getServerSideProps(ctx).props.salesOfferPackage;
            expect(result).toEqual(expectedProps);
        });
    });
});
