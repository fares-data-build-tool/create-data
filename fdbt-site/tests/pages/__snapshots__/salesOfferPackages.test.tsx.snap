// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages salesOfferPackage should render an error when errors are passed through 1`] = `
<Component
  description="Sales Offer Packages page for the Fares Data Build Tool"
  title="Sales Offer Packages - Fares Data Build Tool"
>
  <div
    className="govuk-grid-row"
  >
    <div
      className="govuk-grid-column-two-thirds"
    >
      <CsrfForm
        action="/api/salesOfferPackages"
        csrfToken=""
        method="post"
      >
        <ErrorSummary
          errors={
            Array [
              Object {
                "errorMessage": "error",
                "id": "",
              },
            ]
          }
        />
        <h1
          className="govuk-heading-xl"
        >
          How are the tickets sold - sales offer package
        </h1>
        <span
          className="govuk-hint"
          id="service-list-hint"
        >
          Select all that apply from the lists below
        </span>
        <Component
          errorId="checkbox-0-on-board"
          errors={
            Array [
              Object {
                "errorMessage": "error",
                "id": "",
              },
            ]
          }
        >
          <fieldset
            aria-describedby="sop-purchase-locations"
            className="govuk-fieldset"
          >
            <legend
              className="govuk-fieldset__legend govuk-fieldset__legend--s"
              id="sop-purchase-locations"
            >
              Where can tickets be purchased?
            </legend>
            <FormElementWrapper
              errorClass="govuk-form-group--error"
              errorId="checkbox-0-on-board"
              errors={
                Array [
                  Object {
                    "errorMessage": "error",
                    "id": "",
                  },
                ]
              }
            >
              <div
                className="govuk-checkboxes__item"
                key="checkbox-item-on-board"
              >
                <input
                  className="govuk-checkboxes__input"
                  defaultChecked={false}
                  id="checkbox-0-on-board"
                  name="purchaseLocations"
                  type="checkbox"
                  value="onBoard"
                />
                <label
                  className="govuk-label govuk-checkboxes__label"
                  htmlFor="checkbox-0-on-board"
                >
                  On Board
                </label>
              </div>
              <div
                className="govuk-checkboxes__item"
                key="checkbox-item-online"
              >
                <input
                  className="govuk-checkboxes__input"
                  defaultChecked={false}
                  id="checkbox-1-online"
                  name="purchaseLocations"
                  type="checkbox"
                  value="online"
                />
                <label
                  className="govuk-label govuk-checkboxes__label"
                  htmlFor="checkbox-1-online"
                >
                  Online
                </label>
              </div>
              <div
                className="govuk-checkboxes__item"
                key="checkbox-item-mobile-device"
              >
                <input
                  className="govuk-checkboxes__input"
                  defaultChecked={false}
                  id="checkbox-2-mobile-device"
                  name="purchaseLocations"
                  type="checkbox"
                  value="mobileDevice"
                />
                <label
                  className="govuk-label govuk-checkboxes__label"
                  htmlFor="checkbox-2-mobile-device"
                >
                  Mobile Device
                </label>
              </div>
            </FormElementWrapper>
          </fieldset>
        </Component>
        <Component
          errorId="checkbox-0-cash"
          errors={
            Array [
              Object {
                "errorMessage": "error",
                "id": "",
              },
            ]
          }
        >
          <fieldset
            aria-describedby="sop-payment-methods"
            className="govuk-fieldset"
          >
            <legend
              className="govuk-fieldset__legend govuk-fieldset__legend--s"
              id="sop-payment-methods"
            >
              How can tickets be paid for?
            </legend>
            <FormElementWrapper
              errorClass="govuk-form-group--error"
              errorId="checkbox-0-cash"
              errors={
                Array [
                  Object {
                    "errorMessage": "error",
                    "id": "",
                  },
                ]
              }
            >
              <div
                className="govuk-checkboxes__item"
                key="checkbox-item-cash"
              >
                <input
                  className="govuk-checkboxes__input"
                  defaultChecked={false}
                  id="checkbox-0-cash"
                  name="paymentMethods"
                  type="checkbox"
                  value="cash"
                />
                <label
                  className="govuk-label govuk-checkboxes__label"
                  htmlFor="checkbox-0-cash"
                >
                  Cash
                </label>
              </div>
              <div
                className="govuk-checkboxes__item"
                key="checkbox-item-debit-card"
              >
                <input
                  className="govuk-checkboxes__input"
                  defaultChecked={false}
                  id="checkbox-1-debit-card"
                  name="paymentMethods"
                  type="checkbox"
                  value="debitCard"
                />
                <label
                  className="govuk-label govuk-checkboxes__label"
                  htmlFor="checkbox-1-debit-card"
                >
                  Debit Card
                </label>
              </div>
              <div
                className="govuk-checkboxes__item"
                key="checkbox-item-credit-card"
              >
                <input
                  className="govuk-checkboxes__input"
                  defaultChecked={false}
                  id="checkbox-2-credit-card"
                  name="paymentMethods"
                  type="checkbox"
                  value="creditCard"
                />
                <label
                  className="govuk-label govuk-checkboxes__label"
                  htmlFor="checkbox-2-credit-card"
                >
                  Credit Card
                </label>
              </div>
              <div
                className="govuk-checkboxes__item"
                key="checkbox-item-mobile-phone"
              >
                <input
                  className="govuk-checkboxes__input"
                  defaultChecked={false}
                  id="checkbox-3-mobile-phone"
                  name="paymentMethods"
                  type="checkbox"
                  value="mobilePhone"
                />
                <label
                  className="govuk-label govuk-checkboxes__label"
                  htmlFor="checkbox-3-mobile-phone"
                >
                  Mobile Phone
                </label>
              </div>
            </FormElementWrapper>
          </fieldset>
        </Component>
        <Component
          errorId="checkbox-0-paper-ticket"
          errors={
            Array [
              Object {
                "errorMessage": "error",
                "id": "",
              },
            ]
          }
        >
          <fieldset
            aria-describedby="sop-ticket-formats"
            className="govuk-fieldset"
          >
            <legend
              className="govuk-fieldset__legend govuk-fieldset__legend--s"
              id="sop-ticket-formats"
            >
              What format do the tickets come in?
            </legend>
            <FormElementWrapper
              errorClass="govuk-form-group--error"
              errorId="checkbox-0-paper-ticket"
              errors={
                Array [
                  Object {
                    "errorMessage": "error",
                    "id": "",
                  },
                ]
              }
            >
              <div
                className="govuk-checkboxes__item"
                key="checkbox-item-paper-ticket"
              >
                <input
                  className="govuk-checkboxes__input"
                  defaultChecked={false}
                  id="checkbox-0-paper-ticket"
                  name="ticketFormats"
                  type="checkbox"
                  value="paperTicket"
                />
                <label
                  className="govuk-label govuk-checkboxes__label"
                  htmlFor="checkbox-0-paper-ticket"
                >
                  Paper Ticket
                </label>
              </div>
              <div
                className="govuk-checkboxes__item"
                key="checkbox-item-mobile-app"
              >
                <input
                  className="govuk-checkboxes__input"
                  defaultChecked={false}
                  id="checkbox-1-mobile-app"
                  name="ticketFormats"
                  type="checkbox"
                  value="mobileApp"
                />
                <label
                  className="govuk-label govuk-checkboxes__label"
                  htmlFor="checkbox-1-mobile-app"
                >
                  Mobile App
                </label>
              </div>
              <div
                className="govuk-checkboxes__item"
                key="checkbox-item-smart-card"
              >
                <input
                  className="govuk-checkboxes__input"
                  defaultChecked={false}
                  id="checkbox-2-smart-card"
                  name="ticketFormats"
                  type="checkbox"
                  value="smartCard"
                />
                <label
                  className="govuk-label govuk-checkboxes__label"
                  htmlFor="checkbox-2-smart-card"
                >
                  Smart Card
                </label>
              </div>
            </FormElementWrapper>
          </fieldset>
        </Component>
        <input
          className="govuk-button"
          id="continue-button"
          type="submit"
          value="Continue"
        />
      </CsrfForm>
    </div>
    <div
      className="govuk-grid-column-one-third"
    >
      <Component />
    </div>
  </div>
</Component>
`;

exports[`pages salesOfferPackage should render correctly 1`] = `
<Component
  description="Sales Offer Packages page for the Fares Data Build Tool"
  title="Sales Offer Packages - Fares Data Build Tool"
>
  <div
    className="govuk-grid-row"
  >
    <div
      className="govuk-grid-column-two-thirds"
    >
      <CsrfForm
        action="/api/salesOfferPackages"
        csrfToken=""
        method="post"
      >
        <ErrorSummary
          errors={Array []}
        />
        <h1
          className="govuk-heading-xl"
        >
          How are the tickets sold - sales offer package
        </h1>
        <span
          className="govuk-hint"
          id="service-list-hint"
        >
          Select all that apply from the lists below
        </span>
        <Component
          errorId="checkbox-0-on-board"
          errors={Array []}
        >
          <fieldset
            aria-describedby="sop-purchase-locations"
            className="govuk-fieldset"
          >
            <legend
              className="govuk-fieldset__legend govuk-fieldset__legend--s"
              id="sop-purchase-locations"
            >
              Where can tickets be purchased?
            </legend>
            <FormElementWrapper
              errorClass="govuk-form-group--error"
              errorId="checkbox-0-on-board"
              errors={Array []}
            >
              <div
                className="govuk-checkboxes__item"
                key="checkbox-item-on-board"
              >
                <input
                  className="govuk-checkboxes__input"
                  defaultChecked={false}
                  id="checkbox-0-on-board"
                  name="purchaseLocations"
                  type="checkbox"
                  value="onBoard"
                />
                <label
                  className="govuk-label govuk-checkboxes__label"
                  htmlFor="checkbox-0-on-board"
                >
                  On Board
                </label>
              </div>
              <div
                className="govuk-checkboxes__item"
                key="checkbox-item-online"
              >
                <input
                  className="govuk-checkboxes__input"
                  defaultChecked={false}
                  id="checkbox-1-online"
                  name="purchaseLocations"
                  type="checkbox"
                  value="online"
                />
                <label
                  className="govuk-label govuk-checkboxes__label"
                  htmlFor="checkbox-1-online"
                >
                  Online
                </label>
              </div>
              <div
                className="govuk-checkboxes__item"
                key="checkbox-item-mobile-device"
              >
                <input
                  className="govuk-checkboxes__input"
                  defaultChecked={false}
                  id="checkbox-2-mobile-device"
                  name="purchaseLocations"
                  type="checkbox"
                  value="mobileDevice"
                />
                <label
                  className="govuk-label govuk-checkboxes__label"
                  htmlFor="checkbox-2-mobile-device"
                >
                  Mobile Device
                </label>
              </div>
            </FormElementWrapper>
          </fieldset>
        </Component>
        <Component
          errorId="checkbox-0-cash"
          errors={Array []}
        >
          <fieldset
            aria-describedby="sop-payment-methods"
            className="govuk-fieldset"
          >
            <legend
              className="govuk-fieldset__legend govuk-fieldset__legend--s"
              id="sop-payment-methods"
            >
              How can tickets be paid for?
            </legend>
            <FormElementWrapper
              errorClass="govuk-form-group--error"
              errorId="checkbox-0-cash"
              errors={Array []}
            >
              <div
                className="govuk-checkboxes__item"
                key="checkbox-item-cash"
              >
                <input
                  className="govuk-checkboxes__input"
                  defaultChecked={false}
                  id="checkbox-0-cash"
                  name="paymentMethods"
                  type="checkbox"
                  value="cash"
                />
                <label
                  className="govuk-label govuk-checkboxes__label"
                  htmlFor="checkbox-0-cash"
                >
                  Cash
                </label>
              </div>
              <div
                className="govuk-checkboxes__item"
                key="checkbox-item-debit-card"
              >
                <input
                  className="govuk-checkboxes__input"
                  defaultChecked={false}
                  id="checkbox-1-debit-card"
                  name="paymentMethods"
                  type="checkbox"
                  value="debitCard"
                />
                <label
                  className="govuk-label govuk-checkboxes__label"
                  htmlFor="checkbox-1-debit-card"
                >
                  Debit Card
                </label>
              </div>
              <div
                className="govuk-checkboxes__item"
                key="checkbox-item-credit-card"
              >
                <input
                  className="govuk-checkboxes__input"
                  defaultChecked={false}
                  id="checkbox-2-credit-card"
                  name="paymentMethods"
                  type="checkbox"
                  value="creditCard"
                />
                <label
                  className="govuk-label govuk-checkboxes__label"
                  htmlFor="checkbox-2-credit-card"
                >
                  Credit Card
                </label>
              </div>
              <div
                className="govuk-checkboxes__item"
                key="checkbox-item-mobile-phone"
              >
                <input
                  className="govuk-checkboxes__input"
                  defaultChecked={false}
                  id="checkbox-3-mobile-phone"
                  name="paymentMethods"
                  type="checkbox"
                  value="mobilePhone"
                />
                <label
                  className="govuk-label govuk-checkboxes__label"
                  htmlFor="checkbox-3-mobile-phone"
                >
                  Mobile Phone
                </label>
              </div>
            </FormElementWrapper>
          </fieldset>
        </Component>
        <Component
          errorId="checkbox-0-paper-ticket"
          errors={Array []}
        >
          <fieldset
            aria-describedby="sop-ticket-formats"
            className="govuk-fieldset"
          >
            <legend
              className="govuk-fieldset__legend govuk-fieldset__legend--s"
              id="sop-ticket-formats"
            >
              What format do the tickets come in?
            </legend>
            <FormElementWrapper
              errorClass="govuk-form-group--error"
              errorId="checkbox-0-paper-ticket"
              errors={Array []}
            >
              <div
                className="govuk-checkboxes__item"
                key="checkbox-item-paper-ticket"
              >
                <input
                  className="govuk-checkboxes__input"
                  defaultChecked={false}
                  id="checkbox-0-paper-ticket"
                  name="ticketFormats"
                  type="checkbox"
                  value="paperTicket"
                />
                <label
                  className="govuk-label govuk-checkboxes__label"
                  htmlFor="checkbox-0-paper-ticket"
                >
                  Paper Ticket
                </label>
              </div>
              <div
                className="govuk-checkboxes__item"
                key="checkbox-item-mobile-app"
              >
                <input
                  className="govuk-checkboxes__input"
                  defaultChecked={false}
                  id="checkbox-1-mobile-app"
                  name="ticketFormats"
                  type="checkbox"
                  value="mobileApp"
                />
                <label
                  className="govuk-label govuk-checkboxes__label"
                  htmlFor="checkbox-1-mobile-app"
                >
                  Mobile App
                </label>
              </div>
              <div
                className="govuk-checkboxes__item"
                key="checkbox-item-smart-card"
              >
                <input
                  className="govuk-checkboxes__input"
                  defaultChecked={false}
                  id="checkbox-2-smart-card"
                  name="ticketFormats"
                  type="checkbox"
                  value="smartCard"
                />
                <label
                  className="govuk-label govuk-checkboxes__label"
                  htmlFor="checkbox-2-smart-card"
                >
                  Smart Card
                </label>
              </div>
            </FormElementWrapper>
          </fieldset>
        </Component>
        <input
          className="govuk-button"
          id="continue-button"
          type="submit"
          value="Continue"
        />
      </CsrfForm>
    </div>
    <div
      className="govuk-grid-column-one-third"
    >
      <Component />
    </div>
  </div>
</Component>
`;
